ibis/backends/base/__init__.py

```diff
@@ -703,7 +737,7 @@ def _(path: str, *, backend: str, **kwargs: Any) -> BaseBackend:
 
     Examples
     --------
-    >>> con = ibis.connect("file://relative/path/to/data.duckdb")
+    >>> con = ibis.connect("file:///relative/path/to/data.duckdb")
```

Comment:
Looking at the spec that sqlalchemy bases its implementation on

```
Berners-Lee, Masinter & McCahill                               [Page 14]

RFC 1738            Uniform Resource Locators (URL)        December 1994


   The <wpath> of a WAIS URL consists of the WAIS document-id, encoded
   as necessary using the method described in Section 2.2. The WAIS
   document-id should be treated opaquely; it may only be decomposed by
   the server that issued it.

3.10 FILES

   The file URL scheme is used to designate files accessible on a
   particular host computer. This scheme, unlike most other URL schemes,
   does not designate a resource that is universally accessible over the
   Internet.

   A file URL takes the form:

       file://<host>/<path>

   where <host> is the fully qualified domain name of the system on
   which the <path> is accessible, and <path> is a hierarchical
   directory path of the form <directory>/<directory>/.../<name>.
```

it seems like those other systems might be glossing over that part of the spec.

I interpret a `file:///x` URL as `file://` + `<no host>` + `/` + `x`, which would imply a relative path.

Reply:
The reason this is different here is because I'm using `sa.engine.make_url`.

Reply:
Not maybe the best source, but see https://en.wikipedia.org/wiki/File_URI_scheme. As another reference, Java's stdlib treats file urls as `file:///this/is/an/absolute/path`. Pandas, dask, spark, etc... all also treat these as absolute paths.

Whether we support `file://this/is/a/relative/path` is up for debate, but the triple slash version should definitely be absolute IMO to match other systems.

Reply:
Ah, I think I misinterpreted this as sqlalchemy being correct but it appears to be a sqlalchemy-sim (https://docs.sqlalchemy.org/en/14/dialects/sqlite.html#connect-strings).

It would be swell to allow a relative path. I suppose we can do the usual with file URLs and special case the various file-based databases to work the way we (I?) want them to which is to have `sqlite:///path/to/thing.db` mean absolute and `sqlite://path/to/thing.db` mean relative.

Reply:
I agree that the 2 slash version should be relative for things where the `netloc` doesn't mean anything (and should then be interpreted as part of the path).